tests/sentry/issues/test_utils.py

```diff
@@ -48,3 +58,66 @@ def build_occurrence(self, **overrides: Any) -> IssueOccurrence:
             ]
 
         return IssueOccurrence.from_dict(self.build_occurrence_data(**overrides))
+
+
+class SearchIssueTestMixin(OccurrenceTestMixin):
+    def store_search_issue(
+        self: SnubaTestCase,
+        project_id: int,
+        user_id: int,
+        fingerprints: Sequence[str],
+        environment: Optional[str] = None,
+        insert_time: Optional[datetime] = None,
+    ) -> Tuple[Event, IssueOccurrence, Optional[GroupInfo]]:
+        from sentry.utils import snuba
+
+        insert_timestamp = (insert_time if insert_time else timezone.now()).replace(microsecond=0)
+        user_id_val = f"user_{user_id}"
+
+        event_data = {
+            "tags": [("sentry:user", user_id_val)],
+            "timestamp": iso_format(insert_timestamp),
+        }
+
+        if environment:
+            event_data["environment"] = environment
+            event_data["tags"].extend([("environment", environment)])
+
+        event = self.store_event(
+            data=event_data,
+            project_id=project_id,
+        )
+        occurrence = self.build_occurrence(event_id=event.event_id, fingerprint=fingerprints)
+        saved_occurrence, group_info = save_issue_occurrence(occurrence.to_dict(), event)
```

Comment:
I think we might be better off using `process_event_and_issue_occurrence` here just to keep things as close to how they work in prod as possible.